packages/migrate/src/MigrateEngine.ts

```diff
@@ -51,78 +52,171 @@ export class MigrateEngine {
-    return this.runCommand(this.getRPCPayload('markMigrationApplied', args))
+
+  /**
+   * Create the next migration in the migrations history.
+   * If draft is false and there are no unexecutable steps, it will also apply the newly created migration.
+   * Note: This will use the shadow database on the connectors where we need one.
+   */
+  public createMigration(args: EngineArgs.CreateMigrationInput): Promise<EngineResults.CreateMigrationOutput> {
+    return this.runCommand(this.getRPCPayload('createMigration', args))
   }
-  // Mark an existing failed migration as rolled back in the migrations table. It will still be there, but ignored for all purposes except as audit trail.
-  public markMigrationRolledBack(args: EngineArgs.MarkMigrationRolledBackInput): Promise<void> {
-    return this.runCommand(this.getRPCPayload('markMigrationRolledBack', args))
+
+  /**
+   * Execute a database script directly on the specified live database.
+   * Note that this may not be defined on all connectors.
+   */
+  public dbExecute(args: EngineArgs.DbExecuteInput): Promise<EngineResults.DbExecuteOutput> {
+    return this.runCommand(this.getRPCPayload('dbExecute', args))
+  }
+
+  /**
+   * Make the migration engine panic. Only useful to test client error handling.
+   */
+  public debugPanic(): Promise<void> {
+    return this.runCommand(this.getRPCPayload('debugPanic', undefined))
+  }
+
+  /**
+   * The method called at the beginning of migrate dev to decide the course of action,
+   * based on the current state of the workspace.
+   * It acts as a wrapper around diagnoseMigrationHistory.
+   * Its role is to interpret the diagnostic output,
+   * and translate it to a concrete action to be performed by the CLI.
+   */
+  public devDiagnostic(args: EngineArgs.DevDiagnosticInput): Promise<EngineResults.DevDiagnosticOutput> {
+    return this.runCommand(this.getRPCPayload('devDiagnostic', args))
   }
+
+  /**
+   * Read the contents of the migrations directory and the migrations table, and returns their relative statuses.
+   * At this stage, the migration engine only reads,
+   * it does not write to the database nor the migrations directory, nor does it use a shadow database.
+   */
   public diagnoseMigrationHistory(
     args: EngineArgs.DiagnoseMigrationHistoryInput,
   ): Promise<EngineResults.DiagnoseMigrationHistoryOutput> {
     return this.runCommand(this.getRPCPayload('diagnoseMigrationHistory', args))
   }
-  public planMigration(args: EngineArgs.PlanMigrationInput): Promise<EngineResults.PlanMigrationOutput> {
-    return this.runCommand(this.getRPCPayload('planMigration', args))
+
+  /**
+   * Make sure the migration engine can connect to the database from the Prisma schema.
+   */
+  public ensureConnectionValidity(args: EngineArgs.EnsureConnectionValidityInput): Promise<void> {
+    return this.runCommand(this.getRPCPayload('ensureConnectionValidity', args))
   }
```

Comment:
This is a new method: https://prisma.github.io/prisma-engines/doc/migration_core/json_rpc/index.html#-ensureconnectionvalidity